tests/performance/event_generator/src/event_generator/README.md

```diff
@@ -20,3 +20,91 @@ The library is used by creating instances of the provided generator classes and
+
+# Limitations
+
+## Event volume limitations
+
+### Configured operations limitation
+
+- Each event generator instance is limited by the operations parameter, which specifies the total number of events to generate before stopping.
+- For example, if operations is set to 5, the generator will produce 5 events and then stop. The event volume per generator is therefore capped
+by this parameter.
+
+### System resource limitations
+
+- Disk space: For logEventGenerator (Simulates log file entries with rotation), the total volume of logs generated is limited by disk space.
+If the logs are large or if log rotation occurs frequently, disk space may become a limiting factor.
+
+- File system limits: For syscheckEventGenerator (Simulates file system operations (create, modify, delete)), creating a large number of files
+can hit file system limits.
+
+## Maximum generation rate
+
+- The rate parameter specifies the number of events per second.
+- The start() method in EventGenrator calculates the sleep time between events to maintain the specified rate.
+
+### Processing overhead
+
+- Each event involves processing tasks such as file I/O and data formatting.
+- At very high rates, the processing time per event may exceed the interval calculated for the desired rate, leading to lower actual rates.
+
+### System performance
+
+- High rates may lead to increased CPU usage, potentially causing system slowdowns.
+- Rapid file writes and modifications can saturate disk I/O bandwidth.
+
+## Number of parallel event threads
+
+- Each event generator instance runs in its own thread.
+- The main script initiates and manages these threads.
+
+### Limitations
+
+- System thread limits: The operating system limits the maximum number of threads that can be created.
+- Resource consumption: Each thread consumes system resources (memory and CPU time).
+- Context stwitching overhead: A high number of threads can lead to increased context switching, reducing overall performance.
+
+### Recommendations
+
+- Keep the number of threads at a level that the system can handle efficiently.
+- Monitor system performance to avoid overloading the CPU with excessive threading.
+
+## Other potential limitations
+
+### Disk I/O limitations
+
+- The speed at which the disk can handle read/write operations may become a bottleneck.
+- High-frequency file operations can lead to increased disk latency and I/O wait times.
+- Solid-state drives (SSDs) offer faster I/O performance compared to hard disk drives (HDDs).
+
+### CPU limitations
+
+- The generation of events involves processing that consumes CPU resources.
+- High rates and multiple threads can lead to high CPU utilization, impacting other system operations.
+
+### Memory usage
+
+- Each thread and its associated data structures consume memory.
+- Prolonged operation with many threads may lead to increased memory usage.
```

Comment:
Here’s an improved version of the message:

This documentation is too generic and does not adequately address the specific requirements for tool usage. To improve it, we need precise documentation that highlights real hardware limitations in practical environments. It should include actual data on the maximum EPS (Events Per Second) achieved under various resource configurations. This information will serve as a reference to determine whether the EPS usage in a given environment is feasible.

Reply:
Fixed [here](https://github.com/wazuh/wazuh-qa/pull/5842/files).